bundle/phases/deploy.go

```diff
@@ -176,23 +197,14 @@ func Deploy(ctx context.Context, b *bundle.Bundle, outputHandler sync.OutputHand
 		bundle.ApplyContext(ctx, b, lock.Release(lock.GoalDeploy))
 	}()
 
+	deployPrepare(ctx, b)
+	if logdiag.HasError(ctx) {
+		return
+	}
+
 	bundle.ApplySeqContext(ctx, b,
-		statemgmt.StatePull(),
-		terraform.CheckDashboardsModifiedRemotely(),
-		deploy.StatePull(),
-		mutator.ValidateGitDetails(),
-		terraform.CheckRunningResource(),
 		artifacts.CleanUp(),
-		// libraries.CheckForSameNameLibraries() needs to be run after we expand glob references so we
-		// know what are the actual library paths.
-		// libraries.ExpandGlobReferences() has to be run after the libraries are built and thus this
-		// mutator is part of the deploy step rather than validate.
-		libraries.ExpandGlobReferences(),
-		libraries.CheckForSameNameLibraries(),
-		// SwitchToPatchedWheels must be run after ExpandGlobReferences and after build phase because it Artifact.Source and Artifact.Patched populated
-		libraries.SwitchToPatchedWheels(),
 		libraries.Upload(),
-		trampoline.TransformWheelTask(),
```

Comment:
I think `trampoline.TransformWheelTask()` needs to be run after libraries.Upload because Upload mutates the config to have libraries point to uploaded location and transform needs to use this location

Reply:
Thanks,  good catch. I have moved libraries.Upload() (and artifacts.CleanUp(), seems like it might be relevant as well) to their original place. This affects UX of "bundle plan" but that's ok at the moment, it's only internal command.